web/frontend/src/components/slam.vue

```diff
@@ -95,6 +104,77 @@ const fetchSLAMPose = (name: string): Promise<commonApi.Pose> => {
   });
 };
 
+const executeMoveOnMap = async () => {
+  moveClick = false;
+
+  /*
+   * set request name
+   * here we set the name of the motion service the user is using
+   */
+  motionServiceReq.setName('builtin');
+
+  const value = await fetchSLAMPose(props.name);
+  // set pose in frame
+  const destination = new commonApi.Pose();
+  destination.setX(Math.abs(destinationMarker.x - value.getX()));
+  destination.setY(Math.abs(destinationMarker.z - value.getY()));
+  destination.setZ(value.getZ());
+  destination.setOX(value.getOX());
+  destination.setOY(value.getOY());
+  destination.setOZ(value.getOZ());
+  destination.setTheta(value.getTheta());
+  motionServiceReq.setDestination(destination);
+
+  // set SLAM resource name
+  const slamResourceName = new commonApi.ResourceName();
+  slamResourceName.setNamespace('rdk');
+  slamResourceName.setType('service');
+  slamResourceName.setSubtype('slam');
+  slamResourceName.setName(props.name);
+  motionServiceReq.setSlamServiceName(slamResourceName);
+
+  // set component name
+  const baseResource = filterResources(props.resources, 'rdk', 'component', 'base');
+  if (baseResource === undefined) {
+    toast.error('No base component detected.');
+    return;
+  }
+  const baseResourceName = new commonApi.ResourceName();
+  baseResourceName.setNamespace('rdk');
+  baseResourceName.setType('component');
+  baseResourceName.setSubtype('base');
+  baseResourceName.setName(baseResource[0]!.name);
+  motionServiceReq.setComponentName(baseResourceName);
+  
+  props.client.motionService.moveOnMap(
+    motionServiceReq,
+    new grpc.Metadata(),
+    (error: ServiceError | null, response: motionApi.MoveOnMapResponse | null) => {
+      if (error) {
+        toast.error(`Error moving: ${error}`);
+        return;
+      }
+      toast.success(`MoveOnMap success: ${response!.getSuccess()}`);
+    }
+  );
+
+};
+
+const executeStopMoveOnMap = () => {
```

Comment:
unsure if this is done correctly, here is my inspo:
https://stackoverflow.com/questions/71819450/how-to-cancel-a-server-side-streaming-grpc-call-from-client-side-in-nodejs

Reply:
Nice, I'm glad you found a way to handle this! how does this implementation interact with both closing the UI card, or switching screens?

Reply:
Can we move this moveOnMap to a function that both executeStopOnMap and executeMoveOnMap call the same function with different requests? (Note: if log changes based on below comment, this might be less advisable but could still be gotten around by checking the Req internals, or passing in an extra value)